### PR TITLE
New Serverless Pattern - Invoking Amazon Bedrock Mistral 7B LLM using Java based AWS Lambda

### PR DESCRIPTION
**Pull request by biswanathmukherjee**
*Issue #, if available:*

*Description of changes:*
This sample project deploys an Amazon API Gateway REST API with an AWS Lambda integration. The Lambda function, written in Java, calls the Amazon Bedrock API for Mistral 7B Instruct V0.2 and returns a response containing the generated content.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: apigw-lambda-bedrock-sam-java/src/main/java/com/example/Body.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package com.example;
+
+import org.json.JSONObject;
+
+import java.text.MessageFormat;
+
+
+// record Body(String prompt, double temperature, int top_k, double top_p, int max_tokens_to_sample,
```
Comment (parikhudit) [inline apigw-lambda-bedrock-sam-java/src/main/java/com/example/Body.java:8]: Please remove unneeded code

File: apigw-lambda-bedrock-sam-java/template.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                - "bedrock:PutFoundationModelEntitlement"
+                - "bedrock:ListModelCustomizationJobs"
+                - "bedrock:ListAgents"
+                - "bedrock:ListProvisionedModelThroughputs"
+                - "bedrock:ListCustomModels"
+                - "bedrock:CreateKnowledgeBase"
+                - "bedrock:PutModelInvocationLoggingConfiguration"
+                - "bedrock:ListFoundationModels"
+                - "bedrock:CreateFoundationModelAgreement"
+                - "bedrock:InvokeModel"
```
Comment (parikhudit) [inline apigw-lambda-bedrock-sam-java/template.yaml:53]: You may want to see if there is a scope of optimization with permissions

File: apigw-lambda-bedrock-sam-java/src/lambda-zip.xml
Comment (parikhudit) [inline apigw-lambda-bedrock-sam-java/src/lambda-zip.xml:1]: Do we need this file as a part of commit?

File: apigw-lambda-bedrock-sam-java/README.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+## Deployment Instructions
+
+1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
+   ```bash
+   git clone https://github.com/aws-samples/serverless-patterns
+   ```
+
+2. Change directory to the pattern directory:
+   ```bash
+   cd serverless-patterns/apigw-lambda-bedrock-sam-java-sam
```
Comment (parikhudit) [inline apigw-lambda-bedrock-sam-java/README.md:31]: typo in path

File: apigw-lambda-bedrock-sam-java/example-pattern.json [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      "Based on the given prompt, using Mistral 7B Instruct V0.2 model, Amazon Bedrock generates the content and returns the response to the Lambda function.",
+      "The Lambda function processes the Bedrock API response and sends it back to the user via the API Gateway."
+    ]
+  },
+  "gitHub": {
+    "template": {
+      "repoURL": "https://github.com/aws-samples/serverless-patterns/tree/main/apigw-lambda-bedrock-sam-java",
+      "templateURL": "serverless-patterns/apigw-lambda-bedrock-sam-java",
+      "projectFolder": "apigw-lambda-bedrock-sam-java",
+      "templateFile": "apigw-lambda-bedrock-sam-java/template.yaml"
```
Comment (parikhudit) [inline apigw-lambda-bedrock-sam-java/example-pattern.json:21]: Please remove the projectFolder value from the templateFile

File: apigw-lambda-bedrock-sam-java/example-pattern.json [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+      "Delete the stack: <code>sam delete</code>."
+     
+    ]
+  },
+  "authors": [
+    {
+      "name": "Biswanath Mukherjee",
+      "image": "https://d1rwvjey2iif32.cloudfront.net",
+      "bio": "I am a Sr. Solutions Architect working at AWS India.",
+      "linkedin": "https://www.linkedin.com/in/biswanathmukherjee"
```
Comment (parikhudit) [inline apigw-lambda-bedrock-sam-java/example-pattern.json:58]: just need LinkedIn id

File: apigw-lambda-bedrock-sam-java/example-pattern.json (hunk 4 above)
Comment (biswanathmukherjee) [reply apigw-lambda-bedrock-sam-java/example-pattern.json]: Fixed.

File: apigw-lambda-bedrock-sam-java/README.md (hunk 3 above)
Comment (biswanathmukherjee) [reply apigw-lambda-bedrock-sam-java/README.md]: Fixed

File: apigw-lambda-bedrock-sam-java/src/lambda-zip.xml
Comment (biswanathmukherjee) [reply apigw-lambda-bedrock-sam-java/src/lambda-zip.xml:1]: Yes, this file is required. This creates the zip file with the required dependency.

File: apigw-lambda-bedrock-sam-java/src/main/java/com/example/Body.java (hunk 1 above)
Comment (biswanathmukherjee) [reply apigw-lambda-bedrock-sam-java/src/main/java/com/example/Body.java]: Fixed.

**Pull request merged**